### PR TITLE
Add Realm.deleteFile API [#363]

### LINKED ISSUE #363: Equivalent of deleteRealmFile in JS
Something like in https://github.com/realm/realm-java/issues/834 to delete the realm file.

### PR DESCRIPTION
**Pull request by ashwinphatak**
## What, How & Why?

Reference to the issue(s) addressed by this PR: Fixes #363 

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [x] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: src/node/platform.cpp [hunk 1]
```diff
@@ -134,4 +134,63 @@ void remove_realm_files_from_directory(const std::string &dir_path)
...
+
+    FileSystemRequest delete_req;
+    if (uv_fs_unlink(uv_default_loop(), &delete_req, path.c_str(), nullptr) != 0) {
+        throw UVException(static_cast<uv_errno_t>(delete_req.result));
+    }
+}
+
+void remove_realm_files(const std::string &path)
+{
+    remove_file(path);
```
Comment (kristiandupont) [inline src/node/platform.cpp]: I think this logic could be moved to `RealmClass<T>::delete_file` so we at least don't have to duplicate it internally.

File: src/android/platform.cpp [hunk 2]
```diff
@@ -93,4 +93,27 @@ namespace realm {
                           s_default_realm_directory + "/*.realm.lock";
         system(cmd.c_str());
     }
+
+    static void remove_directory(const std::string &path)
+    {
+        std::string cmd_clear_dir = "rm " + path + "/*";
+        system(cmd_clear_dir.c_str());
```
Comment (kristiandupont) [inline src/android/platform.cpp]: We should change this, but if it's what we already did then let's go with it for now.

File: src/node/platform.cpp [hunk 3]
```diff
@@ -134,4 +134,63 @@ void remove_realm_files_from_directory(const std::string &dir_path)
     }
 }
 
+static void remove_directory(const std::string &path)
+{
+    FileSystemRequest exists_req;
+    if(uv_fs_stat(uv_default_loop(), &exists_req, path.c_str(), nullptr) != 0) {
```
Comment (kneth) [inline src/node/platform.cpp]: Please add a space after `if`

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ X.Y.Z Release notes
 * Added `shouldCompactOnLaunch` to configuration (#507).
 * Added `Realm.compact()` for manually compacting Realm files.
 * Added various methods for permission management (#1204).
+* Added `Realm.deleteFile` for deleting a Realm (#363).
```
Comment (kneth) [inline CHANGELOG.md]: We just did a release so please move this to the next version in the changelog.

File: src/node/platform.cpp (hunk 1 above)
Comment (ashwinphatak) [reply src/node/platform.cpp]: Good point, done.

File: src/android/platform.cpp (hunk 2 above)
Comment (ashwinphatak) [reply src/android/platform.cpp]: Yeah, we're already using 'rm' on Android.

File: src/node/platform.cpp (hunk 3 above)
Comment (ashwinphatak) [reply src/node/platform.cpp]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (ashwinphatak) [reply CHANGELOG.md]: Done

**Comment by ashwinphatak** [comment]
iOS impl is pending due to Xcode setup issues, so not ready to merge yet.

**Comment by ashwinphatak** [comment]
@kneth can you take a quick look again? iOS impl. done, rebased against master, all tests PASS.

**Comment by kneth** [comment]
@ashwinphatak Let's get it merged as soon as possible.

**Pull request merged**